src/runtime/cuda.cpp

```diff
@@ -1228,36 +1227,37 @@ WEAK int halide_cuda_device_and_host_free(void *user_context, struct halide_buff
 WEAK int halide_cuda_wrap_device_ptr(void *user_context, struct halide_buffer_t *buf, uint64_t device_ptr) {
     halide_abort_if_false(user_context, buf->device == 0);
     if (buf->device != 0) {
-        return -2;
+        error(user_context) << "halide_cuda_wrap_device_ptr: device is nonzero";
```

Comment:
suggest changing to "halide_cuda_wrap_device_ptr: device field is already non-zero" to clarify it slightly